core/src/main/java/feast/core/service/SpecService.java

```diff
@@ -299,6 +309,26 @@ public ApplyFeatureSetResponse applyFeatureSet(FeatureSetProto.FeatureSet newFea
       status = Status.UPDATED;
     }
 
+    featureSet.incVersion();
+
+    try {
+      specPublisher
+          .sendDefault(featureSet.getReference(), featureSet.toProto().getSpec())
+          .get(SPEC_PUBLISHING_TIMEOUT_SECONDS, TimeUnit.SECONDS);
+    } catch (Exception e) {
+      throw io.grpc.Status.UNAVAILABLE
+          .withDescription("Unable to publish FeatureSet to Kafka")
```

Comment:
I think we should return the exception message so that users (mostly us) can figure out what is wrong. In the future we can abstract that away.